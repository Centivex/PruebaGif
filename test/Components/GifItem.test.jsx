import { render, screen } from "@testing-library/react";
import GifItem from "../../src/Components/GifItem";

describe('Pruebas en <GiftIten/>',()=>{

    const title = 'Digimon';
    const url = 'https://digimon.com/digimon.jpg'

    test('debe hacer macth con el snapshot', ()=>{
    
       const {container} = render(<GifItem title={title} url = {url}/>);
       expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', ()=>{
        render(<GifItem title={title} url = {url}/>)
        //screen.debug()
        //expect( screen.getByRole('img').src).toBe(url);
        const {src, alt} =  screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test('Debe mostrar el titulo del componente', ()=>{
        render(<GifItem title={title} url = {url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    })
}); 