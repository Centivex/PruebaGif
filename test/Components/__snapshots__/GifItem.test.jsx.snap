// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftIten/> debe hacer macth con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Digimon"
      src="https://digimon.com/digimon.jpg"
    />
    Digimon
  </div>
</div>
`;
